### PR TITLE
fix convert segment task

### PR DESCRIPTION
**Pull request by nishantmonu51**
1) fix serde
2) fix wrong parameter being passed when creating subtask

### CODE REVIEW

File: indexing-service/src/test/java/io/druid/indexing/common/task/ConvertSegmentTaskTest.java [hunk 1]
```diff
@@ -31,16 +33,19 @@
...
   {
     final String dataSource = "billy";
     final Interval interval = new Interval(new DateTime().minus(1000), new DateTime());
 
-    DefaultObjectMapper jsonMapper = new DefaultObjectMapper();
 
     ConvertSegmentTask task = ConvertSegmentTask.create(dataSource, interval, null, false, true);
 
+    String s = jsonMapper.writerWithDefaultPrettyPrinter().writeValueAsString(task);
+    System.out.println(s);
```
Comment (gianm) [inline indexing-service/src/test/java/io/druid/indexing/common/task/ConvertSegmentTaskTest.java]: extraneous println

Comment (nishantmonu51) [inline indexing-service/src/test/java/io/druid/indexing/common/task/ConvertSegmentTaskTest.java]: ah, leftover from debugging, removed

**Comment by drcrallen** [comment]
:+1: after travis

**Comment by gianm** [comment]
:+1:

**Pull request merged**